File: src/main/kotlin/ua/com/lavi/usergenerator/factory/FemaleUserFactory.kt
package ua.com.lavi.usergenerator.factory

/**
 * Created by Oleksandr Loushkin
 */

class FemaleUserFactory : UserFactory() {

    override fun getLastname(): String {
        return randomStringFromList(readResourceLines("femalelastnames.txt"))
    }

    override fun getName(): String {
        return randomStringFromList(readResourceLines("femalenames.txt"))
    }

    override fun getPatronomyc(): String {
        return randomStringFromList(readResourceLines("femalepatronymics.txt"))
    }
}

File: src/main/scala/ua/com/lavi/usergenerator/factory/FemaleUserFactory.scala
package ua.com.lavi.usergenerator.factory

/**
 * Created by Oleksandr Loushkin
 */

class FemaleUserFactory extends UserFactory {

    override def getLastname: String = {
        randomStringFromList(readResourceLines("femalelastnames.txt"))
    }

    override def getName: String = {
        randomStringFromList(readResourceLines("femalenames.txt"))
    }

    override def getPatronomyc: String = {
        randomStringFromList(readResourceLines("femalepatronymics.txt"))
    }
}
